### ISSUE TITLE
Lock on some precomp file stops second repl

### ISSUE THREAD

**Issue by awwaiid** [niner/Inline-Perl5#49]
In window1 I start perl6 repl and do `use Inline::Perl5`. In window2 running perl6 repl hangs. Here are the last two lines from strace showing the file lock:

    open("/home/awwaiid/.perl6/2015.12/precomp/.lock", O_RDWR|O_CREAT|O_CLOEXEC, 0666) = 13
    fcntl(13, F_SETLKW, {type=F_WRLCK, whence=SEEK_SET, start=0, len=0}

I don't know if this is specific to Inline::Perl5, or a rakudo bug, or what. This problem is not encountered for other random modules I used, including Inline::Python and Inline::Perl6.

    This is Rakudo version 2015.12 built on MoarVM version 2015.12
    implementing Perl 6.c.

**Comment by niner** [comment]
That's definitely a rakudo issue, not about Inline::Perl5. It's probably triggered by Inline::Perl5's use of no precompilation;
Does this second repl hang forever or just until the "use Inline::Perl5;" line in the first repl finishes?

**Comment by awwaiid** [comment]
To clarify -- I do `use Inline::Perl5` in window1 perl6-repl, and then try to START the perl6-repl in window2 and it hangs. Window1 perl6-repl is completely fine and responsive. Once I exit the window1 perl6-repl, window2's perl6-repl immediately continues.

I tested with `perl6 -e 'use Inline::Perl5; loop { say "hi" ; sleep 3 }'` in window1 instead of the repl, and in this case window2 repl starts without issue.

I tried a few things with `no precompilation;` and haven't gotten it reproduced outside of Inline::Perl5 yet -- but I think you're right. I think that in the context of the REPL and in some specific setup rakudo isn't letting go of it's precomp lock. I'll try to get a non-Inline::Perl5 reproduction and then close this (in favor of a Rakudobug).
